src/libcollections/btree/set.rs

```diff
@@ -509,6 +509,13 @@ impl<T: Ord> Extend<T> for BTreeSet<T> {
     }
 }
 
+#[unstable(feature = "extend_ref", reason = "recently added")]
+impl<'a, T: 'a + Ord + Copy> Extend<&'a T> for BTreeSet<T> {
```

Comment:
It is indeed, the compiler claims that T may not live long enough at the call to cloned() otherwise.